src/material/list/selection-list.ts

```diff
@@ -322,6 +325,7 @@ export class MatListOption extends _MatListOptionMixinBase implements AfterConte
     'role': 'listbox',
     '[tabIndex]': 'tabIndex',
     'class': 'mat-selection-list mat-list-base',
+    '[class.mat-selection-list-single-select]': '!multiple',
```

Comment:
From what I can tell, this class is only used to style selected items. Maybe we can remove the binding here and update the binding on the item so the class is only applied when selected inside a single-selection list?